### PR TITLE
Activities: Enable Limit on favorites for Mail.

### PR DESCRIPTION
**Pull request by dragotin**
Before it was only possible to limit that for the stream.
Fixes owncloud/enterprise#1430

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -72,3 +72,7 @@ Options -Indexes
 <IfModule pagespeed_module>
   ModPagespeed Off
 </IfModule>
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
+
+ErrorDocument 403 //core/templates/403.php
+ErrorDocument 404 //core/templates/404.php
```
Comment (MorrisJobke) [inline .htaccess]: @dragotin You should not check in this here. Otherwise other users will not have the correct path 😉

Comment (DeepDiver1975) [inline .htaccess]: Indeed

**Comment by PVince81** [comment]
I don't remember why it was limited to streams. I remember there were concerns about file access performance because of complex computations that need to be done to find out whether or not to publish an activity, especially for favorites folders. But I don't see how this would make a difference between stream and email.

**Comment by nickvergessen** [comment]
Emails are created in a background job, where we don't have the users password etc, to mount everything.
Performance was only the second problem.

**Comment by dragotin** [comment]
I fixed the .htaccess f%ckup by force pushing.

**Comment by PVince81** [comment]
> Emails are created in a background job, where we don't have the users password etc, to mount everything.

Okay, so this means if there are external storages with "use the user's password" or so, then the mounts in question cannot be mounted by the background job, so the information about the files stored there cannot be retrieved.

So before this can work in all cases (external storage), more work is needed to make cron jobs be able to use the user's password as stored in the DB by the new auth system in 9.1.

**Comment by VicDeo** [comment]
hmm. The complete solution seems to be a mail queue table with timestamp, recipient and body.
And cronjob will only aggregate this queue by recipient concatenating body and send the resulting emails.

**Comment by PVince81** [comment]
I suppose it might be fine to move forward with this but the UI must have a warning that any external storage mounted with user credentials cannot receive notifications due to technical limitations. (more can be said in the documentation)

Thinking of external storage in general, I think people would anyway expect to receive activities from stuff that happens even outside of ownCloud, which is currently not possible either due to: https://github.com/owncloud/core/issues/26656

**Comment by PVince81** [comment]
Tested, works properly with local storages.

**Comment by PVince81** [comment]
Rebased

**Comment by PVince81** [comment]
Doesn't work with global external storages due to https://github.com/owncloud/activity/issues/568

**Comment by PVince81** [comment]
This will need more extensive testing.

I suggest merging this for 10.0.1 instead and have some QA time scheduled for the many different test cases. Note that we only expect this to work correctly for local storages for the first version.

@pmaier1 @DeepDiver1975 @owncloud/qa

**Pull request merged**